src/main/java/com/rackspace/salus/event/processor/services/StateEvaluator.java

```diff
@@ -17,33 +17,252 @@
+    }
+    if (operand1 == null) {
+      throw new IllegalArgumentException("missing/incorrect type for operand in Metrics: "
+          + expression.getValueName());
+    }
+
+    // Find second operand from comparisionValue and convert to double
+    if (expression.getComparisonValue() instanceof Integer) {
+      operand2 = ((Integer) expression.getComparisonValue()).doubleValue();
+    } else if (expression.getComparisonValue() instanceof Double) {
+      operand2 = (Double)expression.getComparisonValue();
+    } else {
+      throw new IllegalArgumentException("Illegal comparison object for: " +
+          expression.getValueName());
+    }
+    return numericCompareFunctions.get(expression.getComparator()).apply(operand1, operand2);
+  }
+
+  private static boolean getStringValue(ComparisonExpression expression,
+      SalusEnrichedMetric metric) {
+    // Find first operand in metrics and make sure it is string
+    String operand1 = null, operand2;
+    for (Metric m : metric.getMetrics()) {
+      if (m.getName().equals(expression.getValueName())) {
+        if (m.getValueCase() ==  ValueCase.STRING) {
+          operand1 = m.getString();
+        } 
+      }
+    }
+    if (operand1 == null) {
+      throw new IllegalArgumentException("missing/incorrect type for operand in Metrics: " +
+          expression.getValueName());
+    }
+
+    // Find second operand from comparisionValue and make sure it is string
+    if (expression.getComparisonValue() instanceof String) {
+      operand2 = (String)expression.getComparisonValue();
+    } else {
+      throw new IllegalArgumentException("Illegal comparison object for: " +
+          expression.getValueName());
+    }
+    return stringCompareFunctions.get(expression.getComparator()).apply(operand1, operand2);
+  }
+
+  @Data
+  // Generate the synthetic metrics, (currently just percent)
+  static class MetricEvaluator {
+    @NonNull
+    private List<Metric> metrics;
+    Map.Entry<String, Double> eval(MetricExpressionBase node) {
+      Double part = null, total = null;
+      double percentage;
+      PercentageEvalNode percentageEvalNode = (PercentageEvalNode)node;
+      // get the partial value
+      for (Metric metric : metrics) {
+        if (metric.getName().equals(percentageEvalNode.getPart())) {
+          if (metric.getValueCase() == ValueCase.INT) {
+            part = (double) metric.getInt();
+          } else {
+            part = metric.getFloat();
+          }
+        }
+        // get the total value
+        if (metric.getName().equals(percentageEvalNode.getTotal())) {
+          if (metric.getValueCase() == ValueCase.INT) {
+            total = (double) metric.getInt();
+          } else {
+            total = metric.getFloat();
+          }
+        }
+        // calculate the percent
+        if (part != null && total != null) {
+          percentage = part / total * 100;
+          return new AbstractMap.SimpleEntry<>(percentageEvalNode.getAs(), percentage);
```

Comment:
In Java 9 they finally added [`Map.entry(k,v)`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html#entry(K,V)) for this scenario
```suggestion
          return Map.entry(percentageEvalNode.getAs(), percentage);
```